libtrap/src/ifc_file.c

```diff
@@ -301,34 +333,34 @@ int file_recv(void *priv, void *data, uint32_t *size, int timeout)
    if (config->neg_initialized == 0) {
       switch(input_ifc_negotiation((void *) config, TRAP_IFC_TYPE_FILE)) {
       case NEG_RES_FMT_UNKNOWN:
-         VERBOSE(CL_VERBOSE_LIBRARY, "Input_ifc_negotiation result: failed (unknown data format of the output interface).");
+         VERBOSE(CL_VERBOSE_LIBRARY, "FILE INPUT IFC[%u] negotiation result: failed (unknown data format of the output interface).", config->ifc_idx);
          return TRAP_E_FORMAT_MISMATCH;
 
       case NEG_RES_CONT:
-         VERBOSE(CL_VERBOSE_LIBRARY, "Input_ifc_negotiation result: success.");
+         VERBOSE(CL_VERBOSE_LIBRARY, "FILE INPUT IFC[%u] negotiation result: success.", config->ifc_idx);
          config->neg_initialized = 1;
          break;
 
       case NEG_RES_RECEIVER_FMT_SUBSET:
-         VERBOSE(CL_VERBOSE_LIBRARY, "Input_ifc_negotiation result: success (data specifier of the input interface is subset of the output interface data specifier).");
+         VERBOSE(CL_VERBOSE_LIBRARY, "FILE INPUT IFC[%u] negotiation result: success (data specifier of the input interface is subset of the output interface data specifier).", config->ifc_idx);
          config->neg_initialized = 1;
          break;
 
       case NEG_RES_SENDER_FMT_SUBSET:
-         VERBOSE(CL_VERBOSE_LIBRARY, "Input_ifc_negotiation result: success (new data specifier of the output interface is subset of the old one; it was not first negotiation).");
+         VERBOSE(CL_VERBOSE_LIBRARY, "FILE INPUT IFC[%u] negotiation result: success (new data specifier of the output interface is subset of the old one; it was not first negotiation).", config->ifc_idx);
          config->neg_initialized = 1;
          break;
 
       case NEG_RES_FAILED:
-         VERBOSE(CL_VERBOSE_LIBRARY, "Input_ifc_negotiation result: failed (error while receiving hello message from output interface).");
+         VERBOSE(CL_VERBOSE_LIBRARY, "FILE INPUT IFC[%u] negotiation result: failed (error while receiving hello message from output interface).", config->ifc_idx);
          return TRAP_E_FORMAT_MISMATCH;
 
       case NEG_RES_FMT_MISMATCH:
-         VERBOSE(CL_VERBOSE_LIBRARY, "Input_ifc_negotiation result: failed (data format or data specifier mismatch).");
+         VERBOSE(CL_VERBOSE_LIBRARY, "FILE INPUT IFC[%u] negotiation result: failed (data format or data specifier mismatch).", config->ifc_idx);
```

Comment:
Can we somehow print an error message containing something like a diff of templates?

Reply:
Definitely not in file interface, I think it should be in trap.c.